src/app/pages/view/view-records/view-records.component.ts

```diff
@@ -179,13 +184,30 @@ export class ViewRecordsComponent implements OnInit {
         );
     }
 
+    public markAsConsumed() {
+        let params = JSON.parse(JSON.stringify(this.recordParams));
+
+        this.isConsuming = true;
+        this.apiService.getRecordsByDatasetId(this.selectedDataset.id, params).pipe(
+            // map((recordResponse: RecordResponse) => recordResponse.results),
+            mergeMap((records: Record[]) => from(records).pipe(
+                mergeMap((record: Record) => this.apiService.updateRecordConsumed(record.id, true))
```

Comment:
Is this have been merge in core? I don't think so.

Reply:
Yes